buffer_pool_for_uploading.go

```diff
@@ -0,0 +1,27 @@
+package goseaweedfs
+
+import "github.com/oxtoacart/bpool"
+
+// !! For user who wants to use buffer pool when uploading, please explicitly call SetBufferPoolForUploading(poolSize, BufferSize) !!
+
+var gBufferPool *bpool.SizedBufferPool
```

Comment:
I think we could create buffer pool generator our own instead of using additional lib.
For example:

```go
var bufferPool = sync.Pool{
     New: func() interface{} {
         return bytes.NewBuffer(make([]byte, 0, 1<<20)) // Default: 1MB
     },
}

func GetBuffer(expectedSize int) *bytes.Buffer {
   b := bufferPool.Get().(*bytes.Buffer)
   if b.Cap() >= expected {
         b.Reset()
   } else {
         b = bytes.NewBuffer(make([]byte, 0, expected))
   }
   return b
}

func PutBuffer(b *bytes.Buffer) {
   bufferPool.Put(b)
}
```